File: go-geometry/geometry/geometry.go
package geometry

import "fmt"

type Geometry interface {
	Distance(b interface{}) float64
	Scale(factor float64)
}

func PrintDistanceAfterScale(geo1 Geometry, geo2 Geometry) {
	geo1.Scale(2)
	geo2.Scale(2)
	fmt.Println("Distance after scale:", geo1.Distance(geo2))
}

File: rust-geometry/src/geometry.rs
pub trait Geometry {
    fn distance(&self, point_b: &Self) -> f64;
    fn scale(&mut self, factor: i32);
}

pub fn print_distance_after_scale<A: Geometry>(geo1: &mut A, geo2: &mut A) {
    geo1.scale(2);
    geo2.scale(2);
    println!("Distance after scale: {}", geo1.distance(&*geo2));
}
